chore (refs T33878): Add DpTextArea to Storybook. (#380)
demos-europe/demosplan-ui: src/components/DpTextArea/DpTextArea.stories.mdx

import { Canvas, Meta, Story } from '@storybook/addon-docs'
import DpTextArea from './DpTextArea'

<Meta
  title="Components/TextArea"
  component={DpTextArea} />

# DpTextArea

The `DpTextArea` component is used to defines a multi-line text input control.

## Examples

export const Template = (args, { argTypes }) => ({
    props: Object.keys(argTypes),
    components: { DpTextArea },
    template: `<dp-text-area v-bind="$props" />`
});

### Disabled

<Canvas>
    <Story
        name="Disabled"
        args={{
            id: 'textAreaId',

>             label: 'Insert the Text.',

Author: Done in https://github.com/demos-europe/demosplan-ui/pull/380/commits/38ce76b1e88b5d32866319affcbdb62de8676686

            disabled: true
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### Hint

<Canvas>
    <Story
        name="Hint"
        args={{
            id: 'textAreaId',
            label: 'Insert the Text.',
            hint: 'Insert the hint.'
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### growToParent

If enabled, classes are applied to let element grow to 100% height of its parent element.

<Canvas>
    <Story
        name="growToParent"
        args={{
            id: 'textAreaId',
            label: 'Insert the Text.',
            growToParent: true
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### dataDpValidateIf

> Use to conditionally validate a required textarea field.

Reviewer: I think we should explain here what kind of values can be passed to `dataDpValidateIf` so the usage becomes clear.

Author: Done in https://github.com/demos-europe/demosplan-ui/pull/380/commits/ff5e2b7c917e1088c8e3ae991ca71cca46e10afc

<Canvas>
    <Story
        name="dataDpValidateIf"
        args={{
            id: 'textAreaId',
            label: 'Insert the Text.',
            dataDpValidateIf: true
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### reducedHeight

> Display the form field with reduced height (60px).

Reviewer: ```suggestion
Display the textarea with reduced height (60px).
```

<Canvas>
    <Story
        name="reducedHeight"
        args={{
            id: 'textAreaId',
            label: 'Insert the Text.',
            reducedHeight: true
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### Required

> If a TextArea Field is `required`, an asterisk will be added to its label.

Reviewer: ```suggestion
If a textarea is `required`, an asterisk will be added to its label.
```

<Canvas>
    <Story
        name="Required"
        args={{
            id: 'textAreaId',
            label: 'Insert the Text.',
            required: true
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### dataDpValidateErrorFieldname

<Canvas>
    <Story
        name="dataDpValidateErrorFieldname"
        args={{
            id: 'textAreaId',
            label: 'Insert the Text.',

>             dataDpValidateErrorFieldname: 'Text Message here'

Reviewer: ```suggestion
            dataDpValidateErrorFieldname: 'Error message here'
```

        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### Attributes

> Allows passing of additional attributes into the inner form element. Only chosen elements are allowed for each element type, failing if at least one attribute is not allowed within the array.
> Attributes are specified like :attributes="['attrName1=attrValue1', 'attrName2=attrValue2']".
> Params:
> element: String – Type of element

Reviewer: ```suggestion
Allows passing of additional attributes into the inner form element (i.e., the textarea element). Allowed attributes for DpTextArea are
- rows
- cols

If at least one disallowed attribute is passed, an error will be thrown.
Only chosen elements are allowed for each element type, failing if at least one attribute is not allowed within the array.
Attributes are specified as follows:
```

> For Examples:

Reviewer: I'd remove this line

Author: Done in https://github.com/demos-europe/demosplan-ui/pull/380/commits/a5dfd404e9508f80f685043444495e3b8ea7d751

`:attributes="['rows=5', 'cols=10']"`

<Canvas>
    <Story
        name="Attributes"
        args={{
            id: 'textAreaId',

>             label: 'Insert the Text.'

Reviewer: Shouldn't we pass the `attributes` prop here?

Author: Done in https://github.com/demos-europe/demosplan-ui/pull/380/commits/16adc6f860bcd97023bc5bfbd489ae70750486b3

# DpTextArea

The `DpTextArea` component is used to define a multi-line text input control.

## Examples

export const Template = (args, { argTypes }) => ({
    props: Object.keys(argTypes),
    components: { DpTextArea },
    template: `<dp-text-area v-bind="$props" />`
});

### Disabled

<Canvas>
    <Story
        name="Disabled"
        args={{
            id: 'textAreaId',
            label: 'Enter some text',
            disabled: true
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### Hint

<Canvas>
    <Story
        name="Hint"
        args={{
            id: 'textAreaId',
            label: 'Enter some text',
            hint: 'A hint explaining what to enter in the textarea.'
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### growToParent

If enabled, classes are applied to let the element grow to 100% height of its parent element.

<Canvas>
    <Story
        name="growToParent"
        args={{
            id: 'textAreaId',
            label: 'Enter some text',
            growToParent: true
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### dataDpValidateIf

Use to conditionally validate a required textarea field.

> This Prop take values `Boolean` or `String`.
> 
> `data-dp-validate-if="input[name='uploadedFiles[videoSrc]']!=='', #videoTitle!==''"`

Reviewer 2: ```suggestion
A condition consists of an expression that can be passed to `querySelector` (which should target another form element), followed by one of the operators `!==` or `===`, followed by a string to compare against. The operator and value can be omitted if the target element is of type checkbox or radio - in this case the textarea is validated if the referenced radio or checkbox element is checked.

Multiple conditions are separated by a comma.

In this example, a textarea is validated if a checkbox with the id `#send_to_address` is checked:
`data-dp-validate-if="#send_to_address"`

In this example, the textarea is validated if the input with the name `uploadedFiles[videoSrc]` is already filled, but the field with the id `videoTitle` is empty.
`data-dp-validate-if="input[name='uploadedFiles[videoSrc]']!=='', #videoTitle!==''"`
```

<Canvas>
    <Story
        name="dataDpValidateIf"
        args={{
            id: 'textAreaId',
            label: 'Enter some text',
            dataDpValidateIf: true
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### reducedHeight

Display the textarea with reduced height (60px).

<Canvas>
    <Story
        name="reducedHeight"
        args={{
            id: 'textAreaId',
            label: 'Enter some text',
            reducedHeight: true
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### Required

If a textarea is `required`, an asterisk will be added to its label.

<Canvas>
    <Story
        name="Required"
        args={{
            id: 'textAreaId',
            label: 'Enter some text',
            required: true
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

> ### dataDpValidateErrorFieldname
> 
> <Canvas>
>     <Story
>         name="dataDpValidateErrorFieldname"
>         args={{
>             id: 'textAreaId',
>             label: 'Enter some text',
>             dataDpValidateErrorFieldname: 'Error message here'
>         }}>
>         {Template.bind({})}
>     </Story>
> </Canvas>

Reviewer 2: ```suggestion
### dataDpValidateErrorFieldname

Instead of displaying a custom error per field, as achieved by the custom data attribute `data-dp-validate-error` (which is not yet supported in DpTextArea), multiple failing fields may trigger one combined error notification that will then list the field names of each failing field. It reads such as `Bitte korrigieren Sie Ihre Eingabe in folgenden Feldern: Anrede, Betreff, Nachricht. Erst dann können Sie Ihre Angaben speichern.`. If not explicitly set, it defaults to the `label` prop, or, if not set, to null.

<Canvas>
    <Story
        name="dataDpValidateErrorFieldname"
        args={{
            id: 'textAreaId',
            label: 'Your Notes',
            dataDpValidateErrorFieldname: 'Notes'
        }}>
        {Template.bind({})}
    </Story>
</Canvas>
```

import { Canvas, Meta, Story } from '@storybook/addon-docs'

> import DpTextArea from './DpTextArea'

Reviewer 2: We can change this to 

```suggestion
import DpTextArea from './'
```

(see #515)
